authz.go

```diff
@@ -225,6 +237,23 @@ func (a *authorizer) authorize(ctx context.Context, sr authz.SubjectAccessReview
 			break
 		}
 	}
+
+	// if cache and dry run are enabled, error need to be added to log if authorize result from cache and authorize result from zts / zms are different.
+	if a.AuthorizationConfig.Config.UseCache {
+		if decision.status.Allowed != granted {
+			var viaCheck string
+			if decision.via != "" {
```

Comment:
in which case `decision.via` be empty? also, if this could either set from zts flow or cache flow, it would be hard to differentiate from logs. we can append both to the mismatch log message

Reply:
then in this case we need to update the splunk alert query

Reply:
decision.via will be empty if cache.authorize doesn't return true.

Reply:
via can be differentiated by which flow returns true, whoever returns false will set via to empty.

Reply:
sounds good :+1: